dataplane/updater.h

```diff
@@ -38,98 +38,78 @@ namespace dataplane
 }
 
 //
-
-class updater_lpm4_24bit_8bit
+template<typename Address, typename ObjectType>
+class updater_lpm
 {
 public:
-	using object_type = lpm4_24bit_8bit_atomic;
-
-	updater_lpm4_24bit_8bit(const char* name,
-	                        dataplane::memory_manager* memory_manager,
-	                        const tSocketId socket_id) :
+	using stats_t = typename ObjectType::stats_t;
+	updater_lpm(const char* name,
+	            dataplane::memory_manager* memory_manager,
+	            const tSocketId socket_id) :
 	        name(name),
 	        memory_manager(memory_manager),
 	        socket_id(socket_id),
-	        pointer(nullptr)
+	        pointer_{nullptr, memory_manager->deleter()}
 	{
 		stats.extended_chunks_count = 0;
-		stats.extended_chunks_size = 0;
 		stats.max_used_chunk_id = 0;
 		stats.free_chunk_cache.flags = 0;
 	}
 
 	eResult init()
 	{
-		return create(2 * object_type::extended_chunks_size_min);
-	}
-
-	eResult create(const uint64_t extended_chunks_size)
-	{
-		auto* next_pointer = memory_manager->create<object_type>(name.data(),
-		                                                         socket_id,
-		                                                         object_type::calculate_sizeof(extended_chunks_size));
-		if (next_pointer == nullptr)
+		auto default_chunks_size = 2 * ObjectType::extended_chunks_size_min;
+		pointer_ = memory_manager->create_unique<ObjectType>(name.c_str(),
+		                                                     socket_id,
+		                                                     ObjectType::calculate_sizeof(default_chunks_size));
+		if (pointer_)
 		{
-			return eResult::errorAllocatingMemory;
+			stats.extended_chunks_size = default_chunks_size;
+			return eResult::success;
 		}
-
-		object_type::stats_t next_stats;
-		next_stats.extended_chunks_count = 0;
-		next_stats.extended_chunks_size = extended_chunks_size;
-		next_stats.max_used_chunk_id = 0;
-		next_stats.free_chunk_cache.flags = 0;
-
-		if (pointer)
+		else
 		{
-			next_pointer->copy(next_stats, stats, *pointer);
-			memory_manager->destroy(pointer);
+			return eResult::errorAllocatingMemory;
 		}
-
-		stats = next_stats;
-		pointer = next_pointer;
-
-		return eResult::success;
 	}
```

Comment:
Let's do it like in `updater_array` to be consistent, i.e reintroduce the `create` method to allocate memory and set up the `pointer_`, and here just do
```cpp
    eResult init() {
        return create(2 * ObjectType::extended_chunks_size_min);
    }
```

Reply:
Ok, we don't want to reintroduce the `create` since it has some hidden logic of copying data, but we sill have some code duplication with `init` and `Resize` methods. Maybe some other private method (not `create`) will do the job?

Reply:
How about
```cpp
	memory_manager::unique_ptr<ObjectType> Allocate(const std::size_t size)
	{
		return memory_manager->create_unique<ObjectType>(name.c_str(),
		                                                 socket - id,
		                                                 ObjectType::calculate_sizeof(size));
	}
```
Everything else is specific to `init` and `Resize`

Reply:
`updater_array` is different not only in it's "arrayness", but also doesn't track stats. IMHO naming similarity is unfortunate coincidence.